packages/zk-eddsa-event-ticket-pcd/test/ZKEdDSAEventTicketPCD.spec.ts

```diff
@@ -0,0 +1,394 @@
+    for (let i = 0; i < input.length; i++) {
+      expected[i] = uuidToBigInt(input[i]).toString();
+    }
+    expect(snarkInputForValidEventIds(input)).deep.equal(expected);
+  });
+  it("should handle full input", async function () {
+    const input: string[] = [];
+    const expected: string[] = [];
+    for (let i = 0; i < VALID_EVENT_IDS_MAX_LEN; i++) {
+      const id = uuid();
+      input.push(id.toString());
+      expected.push(uuidToBigInt(id.toString()).toString());
+    }
+    expect(snarkInputForValidEventIds(input)).deep.equal(expected);
+  });
+  it("should reject input if too large", async function () {
+    const input: string[] = [];
+    const expected: string[] = [];
+    for (let i = 0; i < VALID_EVENT_IDS_MAX_LEN + 1; i++) {
+      const id = uuid();
+      input.push(id.toString());
+      expected.push(uuidToBigInt(id.toString()).toString());
+    }
+    assert.throws(() => {
+      snarkInputForValidEventIds(input);
+    });
+  });
+});
+
+describe("ZKEdDSAEventTicketPCD should work", function () {
+  this.timeout(1000 * 30);
+
+  let pcd1: ZKEdDSAEventTicketPCD;
+
+  const ticketData1: ITicketData = {
+    // the fields below are not signed and are used for display purposes
+
+    attendeeName: "test name",
+    attendeeEmail: "user@test.com",
+    eventName: "event",
+    ticketName: "ticket",
+    checkerEmail: "checker@test.com",
+
+    // the fields below are signed using the server's private eddsa key
+
+    ticketId: "b38501b0-cf66-4416-a4ef-631a9a9b32c4",
+    eventId: "4332f16c-8444-4261-94b9-a0ba8ca917e2",
+    productId: "c94002fb-2c41-480b-842d-fa826fb517e5",
+    timestampConsumed: 0,
+    timestampSigned: 1693028498280,
+    attendeeSemaphoreId: identity1.getCommitment().toString(),
+    isConsumed: false,
+    isRevoked: false
+  };
+
+  const fieldsToReveal1: EdDSATicketFieldsToReveal = {
+    revealEventId: true,
+    revealProductId: true
+  };
+
+  const fieldsToReveal2: EdDSATicketFieldsToReveal = {
+    revealEventId: true,
+    revealProductId: true,
+    revealIsConsumed: true,
+    revealIsRevoked: true
+  };
+
+  // validEventIds1 contains ticketData1.eventId
+  const validEventIds1: string[] = [
+    uuid(),
+    ticketData1.eventId,
+    uuid(),
+    uuid()
+  ];
+
+  // validEventIds2 does not contain ticketData1.eventId
+  const validEventIds2: string[] = [uuid(), uuid(), uuid(), uuid()];
+
+  async function makeSerializedIdentityPCD(
+    identity: Identity
+  ): Promise<SerializedPCD<SemaphoreIdentityPCD>> {
+    const identityPCD = await SemaphoreIdentityPCDPackage.prove({
+      identity: identity
+    });
+
+    return await SemaphoreIdentityPCDPackage.serialize(identityPCD);
+  }
+
+  this.beforeAll(async function () {
+    await EdDSATicketPCDPackage.init?.({});
+    if (!ZKEdDSAEventTicketPCDPackage.init) return;
+    await ZKEdDSAEventTicketPCDPackage.init({
+      zkeyFilePath,
+      wasmFilePath
+    });
+
+    toArgs = async (
```

Comment:
any reason you're defining this not where you're declaring it? if it's because you want to initialize the packages first, that's not necessary.

Reply:
This was copy/pasted from @gubsheep's unit tests, so he'd have to say what he had in mind.
I think this pattern might've been necessary if the lambda was capturing outputs from initialization, but since it's not I think the function can be safely moved up.

Reply:
hm, i did it for the reason ivan mentioned, iirc i was running into some sort of error when i defined it when i declared it above. but i don't remember super well now and am happy to defer to you guys to just do whatever works.